cnd/src/swap_protocols/rfc003/bob/actions/erc20.rs

```diff
@@ -50,19 +50,19 @@ where
 
         use self::LedgerState::*;
 
-        let mut actions = match (alpha_state, beta_state, self.secret) {
-            (Funded { htlc_location, .. }, _, Some(secret)) => {
+        let mut actions = match (alpha_state, beta_state) {
+            (Funded { htlc_location, .. }, Redeemed { secret, .. }) => {
```

Comment:
Bob can now access the funding by inspecting the `beta_ledger_state`. In case that one is `Redeemed`, the secret used by Alice is captured in there.